### ISSUE TITLE
Heapster pod not started in 600s flake

### ISSUE THREAD

**Issue by goltermann** [kubernetes/kubernetes#24483]
In a GCE e2e test the submit queue ran for me, heapster never started.

The containers that are scheduled as part of the pod never come up ready.  I see the pod scheduled, the image pull start, but nothing comes up and the suite times out.

Selected log lines:
```
Apr 18 23:15:31.628: INFO: Waiting up to 10m0s for all pods (need at least 1) in namespace 'kube-system' to be running and ready
Apr 18 23:15:31.670: INFO: The status of Pod heapster-v1.1.0.beta1-2006342526-1keqz is Pending, waiting for it to be either Running or Failed
Apr 18 23:15:31.671: INFO: heapster-v1.1.0.beta1-2006342526-1keqz              e2e-gce-master-0-minion-55ql  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-04-18 23:13:48 -0700 PDT ContainersNotReady containers with unready status: [heapster eventer heapster-nanny eventer-nanny]}]
<loops for ten minutes, then fails>
```

Events:
```
Apr 18 23:25:31.785: INFO: At 2016-04-18 23:13:16 -0700 PDT - event for heapster-v1.1.0.beta1: {deployment-controller } ScalingReplicaSet: Scaled up replica set heapster-v1.1.0.beta1-2006342526 to 1
Apr 18 23:25:31.785: INFO: At 2016-04-18 23:13:16 -0700 PDT - event for heapster-v1.1.0.beta1-2006342526: {replicaset-controller } SuccessfulCreate: Created pod: heapster-v1.1.0.beta1-2006342526-1keqz
Apr 18 23:25:31.785: INFO: At 2016-04-18 23:13:16 -0700 PDT - event for heapster-v1.1.0.beta1-2006342526-1keqz: {default-scheduler } FailedScheduling: no nodes available to schedule pods
Apr 18 23:25:31.785: INFO: At 2016-04-18 23:13:47 -0700 PDT - event for heapster-v1.1.0.beta1-2006342526-1keqz: {default-scheduler } Scheduled: Successfully assigned heapster-v1.1.0.beta1-2006342526-1keqz to e2e-gce-master-0-minion-55ql
Apr 18 23:25:31.785: INFO: At 2016-04-18 23:13:50 -0700 PDT - event for heapster-v1.1.0.beta1-2006342526-1keqz: {kubelet e2e-gce-master-0-minion-55ql} Pulling: pulling image "gcr.io/google_containers/heapster:v1.1.0-beta1"
Apr 18 23:25:31.803: INFO: heapster-v1.1.0.beta1-2006342526-1keqz              e2e-gce-master-0-minion-55ql  Pending         [{Ready False 0001-01-01 00:00:00 +0000 UTC 2016-04-18 23:13:48 -0700 PDT ContainersNotReady containers with unready status: [heapster eventer heapster-nanny eventer-nanny]}]
```
Apr 18 23:25:32.350: INFO: 
Logging pods the kubelet thinks is on node e2e-gce-master-0-minion-55ql
<snip>
```
Apr 18 23:25:32.559: INFO: heapster-v1.1.0.beta1-2006342526-1keqz started at 2016-04-18 23:13:48 -0700 PDT (4 container statuses recorded)
Apr 18 23:25:32.559: INFO: 	Container eventer ready: false, restart count 0
Apr 18 23:25:32.559: INFO: 	Container eventer-nanny ready: false, restart count 0
Apr 18 23:25:32.559: INFO: 	Container heapster ready: false, restart count 0
Apr 18 23:25:32.559: INFO: 	Container heapster-nanny ready: false, restart count 0
```

**Comment by dchen1107** [comment]
@goltermann Based on the events you posted, looks like the image is never successfully pulled. Or you filtered the related events.

**Comment by dchen1107** [comment]
Do you still have Kubelet logs somewhere?

**Comment by goltermann** [comment]
Yeah, it's this run http://pantheon/storage/browser/kubernetes-jenkins/pr-logs/pull/24234/kubernetes-pull-build-test-e2e-gce/36005/artifacts/e2e-gce-master-0-minion-55ql/

**Comment by bprashanth** [comment]
Turns out only 4 pods actually reached the pulled image state from those kubelet logs:
```
I0419 06:13:56.035568    3416 server.go:610] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"kubernetes-dashboard-v1.0.1-9btlh", UID:"cdd2e0e9-05f5-11e6-80f7-42010af00002", APIVersion:"v1", ResourceVersion:"126", FieldPath:"spec.containers{kubernetes-dashboard}"}): type: 'Normal' reason: 'Pulled' Successfully pulled image "gcr.io/google_containers/kubernetes-dashboard-amd64:v1.0.1"
I0419 06:14:20.917225    3416 server.go:610] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"kube-proxy-e2e-gce-master-0-minion-55ql", UID:"6af60127056076392f77a7376c890e0c", APIVersion:"v1", ResourceVersion:"", FieldPath:"spec.containers{kube-proxy}"}): type: 'Normal' reason: 'Pulled' Container image "gcr.io/google_containers/kube-proxy:bf67fb335dd7899009cff05762b588ba" already present on machine
I0419 06:16:08.028471    3416 server.go:610] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"elasticsearch-logging-v1-grlww", UID:"ce2020dc-05f5-11e6-80f7-42010af00002", APIVersion:"v1", ResourceVersion:"128", FieldPath:"spec.containers{elasticsearch-logging}"}): type: 'Normal' reason: 'Pulled' Successfully pulled image "gcr.io/google_containers/elasticsearch:1.8"
I0419 06:16:15.100347    3416 server.go:610] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"fluentd-elasticsearch-e2e-gce-master-0-minion-55ql", UID:"e313182cba6f0619720f91d8860ae1bb", APIVersion:"v1", ResourceVersion:"", FieldPath:"spec.containers{fluentd-elasticsearch}"}): type: 'Normal' reason: 'Pulled' Successfully pulled image "gcr.io/google_containers/fluentd-elasticsearch:1.15"
```

So I'm guessing this is image pulling again, but I didn't follow the heapster pod to make sure.

**Comment by fejta** [comment]
Assigning to @dchen1107 per team/node label

**Comment by dchen1107** [comment]
Yes, pulling heapster image is never finished since there is only pulling event, no corresponding pulled event. 

```
I0419 06:13:50.238014    3416 server.go:610] Event(api.ObjectReference{Kind:"Pod", Namespace:"kube-system", Name:"heapster-v1.1.0.beta1-2006342526-1keqz", UID:"ce4dd92d-05f5-11e6-80f7-42010af00002", APIVersion:"v1", ResourceVersion:"135", FieldPath:"spec.containers{heapster}"}): type: 'Normal' reason: 'Pulling' pulling image "gcr.io/google_containers/heapster:v1.1.0-beta1"
```

Closed it as dup of #25277
